### PR TITLE
Wire up the components with main

### PR DESCRIPTION
**Pull request by srikanthccv**
Part of https://github.com/SigNoz/signoz/issues/1786

See first https://github.com/SigNoz/signoz-otel-collector/pull/57

This PR wires up the service package with the main package, make all components glued up, and enables the collector to run as a remote-controlled agent.

### CODE REVIEW

File: cmd/signozcollector/main.go [hunk 1]
```diff
@@ -1,43 +1,96 @@
...
-	params := service.CollectorSettings{
-		Factories: factories,
-		BuildInfo: info,
+	ctx := context.Background()
+
+	coll := signozcol.New(
+		signozcol.WrappedCollectorSettings{
+			ConfigPaths: []string{collectorConfig},
+			// TODO: Build version from git tag
+			Version:     "0.63.0",
```
Comment (nityanandagohain) [inline cmd/signozcollector/main.go:51]: @srikanthccv  do we need to add this now?

Comment (nityanandagohain) [reply cmd/signozcollector/main.go:51]: Can you please add then changes? we can merge it after that.

Comment (srikanthccv) [reply cmd/signozcollector/main.go:51]: The deployment patten is not yet clear in k8s, so I would hold it till we figure it out.

Comment (nityanandagohain) [reply cmd/signozcollector/main.go:51]: Got it

**Comment by srikanthccv** [comment]
@rkssisodiya This has the following changes besides the piecing together OpAMP with the main

- wait indefinitely until the collector receives the first config
- restart the collector even if the config hasn’t changed for the first time (the OpAMP server may not have anything new to offer)
- handle some async errors from the collector service
- make OpAMP optional when the manager config (i.e Server URL) is not provided

File: cmd/signozcollector/Dockerfile.old [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Builder stage
```
[the remark this answers is not in the document]
Comment (srikanthccv) [reply cmd/signozcollector/Dockerfile.old:1]: This is not needed. I will remove.

File: Makefile [hunk 3]
```diff
@@ -29,7 +29,7 @@ test-and-lint: test fmt lint
 
 .PHONY: test
 test:
-	go test -count=1 -v -race -cover ./...
+	go test --timeout=30s -count=1 -v -race -cover ./...
```
[the remark this answers is not in the document]
Comment (srikanthccv) [reply Makefile:32]: I don't remember why I added this. Since there is default 1 minute I don't think this is needed.

**Comment by srikanthccv** [comment]
@rkssisodiya I resolved the comments and did some cleanup. Please take another look. This is also good for testing/using. Let me know if there are any issues.

**Pull request merged**